### ISSUE TITLE
unused local not detected on recursive functions

### ISSUE THREAD

**Issue by DetachHead** [microsoft/TypeScript#45360]
# Bug Report

### 🔎 Search Terms

unused local recursive function

### 🕗 Version & Regression Information

v4.5.0-dev.20210807

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?noUncheckedIndexedAccess=true&noUnusedLocals=true&noUnusedParameters=true&jsx=0&module=1&noFallthroughCasesInSwitch=true&noImplicitOverride=true&noPropertyAccessFromIndexSignature=true&ts=4.5.0-dev.20210807#code/KYDwDg9gTgLgBAbwL4FgBQ6D0mB2E7BRTQEBuwOcMAFhAK4Dm1cAZhPgJYDOcOw5UOFGABDACboAxhBxd4bfAF44ACgCUcRQD5E6OPtbt16VBjTZCxKAC44AcgBGIqHbjc4Y4JIA2z4GLgHOngOGB5SEW86YDcePgEhUTEAOhUANgBGAGYstSkZOUDnTVUNbUQkIA)

### 💻 Code

```ts
export {}

//no error even though foo is never read
const foo = () => {
    foo()
}

//error: 'bar' is declared but its value is never read.(6133)
const bar = () => {}
```

### 🙁 Actual behavior

no error even though foo is never read

### 🙂 Expected behavior

```
'foo' is declared but its value is never read.(6133)
```

**Comment by anasdevv** [comment]
Hi there! I hope you're doing well. I came across this issue and was wondering if it's still open? I'd love to contribute to it as I'm new to open source and eager to get involved. Please let me know if it's available. Thank you!

**Comment by DetachHead** [comment]
yes this issue is still present in the latest nightly (5.8.0-dev.20241110)
